components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/SyncManager.kt

```diff
@@ -150,12 +155,22 @@ internal abstract class SyncManager(
      *
      * @param reason A [SyncReason] indicating why this sync is being requested.
      * @param debounce Whether or not this sync should debounced.
+     * @param customEngineSubset A subset of supported engines to sync. Defaults to all supported engines.
      */
-    internal fun now(reason: SyncReason, debounce: Boolean = false) = synchronized(this) {
+    internal fun now(
+        reason: SyncReason,
+        debounce: Boolean = false,
+        customEngineSubset: List<SyncEngine> = listOf(),
+    ) = synchronized(this) {
         if (syncDispatcher == null) {
             logger.info("Sync is not enabled. Ignoring 'sync now' request.")
         }
-        syncDispatcher?.syncNow(reason, debounce)
+        if (!syncConfig.supportedEngines.containsAll(customEngineSubset)) {
```

Comment:
Personally I would have done this check in `FxaAccountManager#syncNow` to avoid going through all of its logic  and doing an additional function call. You have a `syncConfig` there.